### PR TITLE
[SourceTree] Set root directory

### PR DESCRIPTION
**Pull request by khal0988**
Associated Issue: #4121

### Summary of Changes

* highlight the root node in the UI
* Right click would have a context menu option to "reset the directory"

### Test Plan

will test when redux work is done

Example test plan:
- TBD

### Screenshots
screen shot 2017-10-12 at 5 57 31 pm

### CODE REVIEW

File: src/components/PrimaryPanes/SourcesTree.js [hunk 1]
```diff
@@ -142,9 +161,15 @@ class SourcesTree extends Component {
...
+        addToTree(
+          uncollapsedTree,
+          source,
+          this.props.debuggeeUrl,
+          this.props.projectRoot
+        );
       }
-      sourceTree = collapseTree(uncollapsedTree);
+      const unsortedTree = collapseTree(uncollapsedTree);
+      sourceTree = sortEntireTree(unsortedTree, nextProps.debuggeeUrl);
```
Comment (jasonLaster) [inline src/components/PrimaryPanes/SourcesTree.js]: we shouldn't change this...

File: assets/panel/debugger.properties [hunk 2]
```diff
@@ -24,6 +24,11 @@ copySource.accesskey=y
 copySourceUri2=Copy source URI
 copySourceUri2.accesskey=u
 
+# LOCALIZATION NOTE (setDirectoryRoot): This is the text that appears in the
```
Comment (flodolo) [inline assets/panel/debugger.properties]: setDirectoryRoot -> setDirectoryRoot.label

File: assets/panel/debugger.properties [hunk 3]
```diff
@@ -24,6 +24,11 @@ copySource.accesskey=y
...
+# context menu to set a directory as root directory
+setDirectoryRoot.label= Set directory root
```
Comment (flodolo) [inline assets/panel/debugger.properties]: This is just me nitpicking, but please leave no space before and after = (or leave one both ends).

File: assets/panel/debugger.properties (hunk 2 above)
Comment (juliandescottes) [reply assets/panel/debugger.properties]: @jasonLaster this was not fixed before merging!

Comment (jasonLaster) [reply assets/panel/debugger.properties]: oops - sorry i missed this :/

**Pull request merged**